java/bulk-import/bulk-import-runner/src/test/java/sleeper/bulkimport/job/runner/rdd/SingleFileWritingIteratorTest.java

```diff
@@ -153,18 +103,169 @@ public void shouldWriteAllRecordsToAParquetFile() {
         ).iterator();
 
         // When
-        SingleFileWritingIterator fileWritingIterator = new SingleFileWritingIterator(input,
-                createInstanceProperties(), createTableProperties(),
-                new Configuration(), getPartitionTree());
+        SingleFileWritingIterator fileWritingIterator = createIteratorOverRecords(input);
 
         // Then
         assertThat(fileWritingIterator).toIterable()
-                .extracting(row -> readRecords(row.getString(1)))
+                .extracting(row -> readRecords(readPathFromOutputFileMetadata(row)))
                 .containsExactly(
                         Arrays.asList(
                                 createRecord("a", 1, 2),
                                 createRecord("b", 1, 2),
                                 createRecord("c", 1, 2),
                                 createRecord("d", 1, 2)));
     }
+
+    @Test
+    void shouldOutputMetadataPointingToSingleFileInFolderForPartition() {
+        // Given
+        Iterator<Row> input = Lists.newArrayList(
+                RowFactory.create("a", 1, 2),
+                RowFactory.create("b", 1, 2),
+                RowFactory.create("c", 1, 2),
+                RowFactory.create("d", 1, 2)
+        ).iterator();
+        PartitionTree partitionTree = createPartitionsBuilder()
+                .singlePartition("test-partition")
+                .buildTree();
+
+        // When
+        SingleFileWritingIterator fileWritingIterator = createIteratorOverRecordsWithPartitionsAndOutputFilename(
+                input, partitionTree, "test-file");
+
+        // Then
+        assertThat(fileWritingIterator).toIterable()
+                .containsExactly(RowFactory.create(
+                        "test-partition",
+                        "file://" + tempFolder + "/partition_test-partition/test-file.parquet",
+                        4));
+    }
```

Comment:
We could gather these tests into a nested class with common setup like the one below.